src/storage/patch/RdfImmutableCheckPatcher.ts

```diff
@@ -0,0 +1,75 @@
+import type { Quad } from 'rdf-js';
+import type { AuxiliaryStrategy } from '../../http/auxiliary/AuxiliaryStrategy';
+import { BasicRepresentation } from '../../http/representation/BasicRepresentation';
+import type { Representation } from '../../http/representation/Representation';
+import { getLoggerFor } from '../../logging/LogUtil';
+import { INTERNAL_QUADS } from '../../util/ContentTypes';
+import { ConflictHttpError } from '../../util/errors/ConflictHttpError';
+import { serializeQuads, uniqueQuads } from '../../util/QuadUtil';
+import { cloneRepresentation } from '../../util/ResourceUtil';
+import { readableToQuads, readableToString } from '../../util/StreamUtil';
+import { RDF, PIM, LDP } from '../../util/Vocabularies';
+import type { RepresentationPatcherInput } from './RepresentationPatcher';
+import { RepresentationPatcher } from './RepresentationPatcher';
+
+export class RdfImmutableCheckPatcher extends RepresentationPatcher {
+  protected readonly logger = getLoggerFor(this);
+
+  private readonly patcher: RepresentationPatcher;
+  private readonly metaStrategy: AuxiliaryStrategy;
+
+  public constructor(patcher: RepresentationPatcher, metaStrategy: AuxiliaryStrategy) {
+    super();
+    this.patcher = patcher;
+    this.metaStrategy = metaStrategy;
+  }
+
+  public async handle(input: RepresentationPatcherInput): Promise<Representation> {
+    let { representation } = input;
+    const { identifier } = input;
+
+    representation = representation ?? new BasicRepresentation([], INTERNAL_QUADS);
+    const immutableTriples: [string, string | undefined][] = [];
+    immutableTriples.push([ RDF.type, PIM.Storage ]);
+    immutableTriples.push([ LDP.contains, undefined ]);
+
+    let patched: Representation;
+    if (this.metaStrategy.isAuxiliaryIdentifier(identifier)) {
```

Comment:
Everything in this if-statement should be in a separate statement to make the if/else more readable.

But I would actually change this class so it has a `canHandle` that throws an error if the target is not a metadata resource so this if-statement is not needed. The patcher config can then be changed to have a waterfall handler of this patcher and the standard rdf patcher we have there now.